src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs

```diff
@@ -471,6 +472,28 @@ public List<PackageManagerSearchElement> ListAll()
             return CachedPackageList;
         }
 
+        /// <summary>
+        /// Returns a dictionary of infected package(s) with name and version, if the last published version of package uploaded by the current user was flagged as infected.
+        /// </summary>
+        /// <returns></returns>
+        public List<PackageManagerSearchElement> GetInfectedPackages()
+        {
+            InfectedPackageList = new List<PackageManagerSearchElement>();
+            var latestPkgs = Model.GetUsersLatestPackages();
+            if (latestPkgs != null && latestPkgs.maintains.Count > 0)
+            {
+                foreach (var infectedVer in latestPkgs.maintains)
```

Comment:
Would you remind me why do we need to look at each version here? I thought we only need to check the latest version of each package

Reply:
The API will return all packages with last uploaded version, so 1 version per package only.